### PR TITLE
[seastar] Add new port

### PR DESCRIPTION
**Pull request by Balnian**
**Describe the pull request**
Adds a port for seastar framework
- What does your PR fix? Fixes #
Adds a port for seastar
- Which triplets are supported/not supported? Have you updated the CI baseline?
x64-linux is supported
I'll probably have to update CI baseline because this port requires external libraries and tools not available through vcpkg
- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
yes?

### CODE REVIEW

**Comment by JackBoosY** [comment]
Regression with `x64-linux`:
```
CMake Error at /mnt/vcpkg-ci/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find GnuTLS: Found unsuitable version "", but required is at
  least "3.3.26" (found GnuTLS_LIBRARY-NOTFOUND)
Call Stack (most recent call first):
  /mnt/vcpkg-ci/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:580 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindGnuTLS.cmake:45 (find_package_handle_standard_args)
  /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:646 (_find_package)
  cmake/SeastarDependencies.cmake:107 (find_package)
  CMakeLists.txt:309 (seastar_find_dependencies)
```

**Comment by Balnian** [comment]
This port requires the user to install some package from its Linux package manager.
How should this be handled/expressed in the CI? (Add an entry in ci.baseline.txt? Adding the required package in the CI runner image?)

File: ports/seastar/vcpkg.json [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "seastar",
+  "version-string": "20.05.0",
+  "description": "High performance server-side application framework",
+  "homepage": "https://github.com/scylladb/seastar",
+  "documentation": "http://docs.seastar.io/master/index.html",
+  "license": "Apache-2.0",
+  "dependencies": [
+    "boost",
```
Comment (JackBoosY) [inline ports/seastar/vcpkg.json:9]: Can you refine the components of boost instead of filling in the entire boost here?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/seastar/vcpkg.json (hunk 1 above)
Comment (Balnian) [reply ports/seastar/vcpkg.json]: This has been addressed in my latest commit

**Comment by JackBoosY** [comment]
When building `seastar:x64-linux`:
```
CMake Error at /mnt/vcpkg-ci/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:218 (message):
  Could NOT find GnuTLS: Found unsuitable version "", but required is at
  least "3.3.26" (found GnuTLS_LIBRARY-NOTFOUND)
Call Stack (most recent call first):
  /mnt/vcpkg-ci/downloads/tools/cmake-3.19.2-linux/cmake-3.19.2-Linux-x86_64/share/cmake-3.19/Modules/FindPackageHandleStandardArgs.cmake:580 (_FPHSA_FAILURE_MESSAGE)
  cmake/FindGnuTLS.cmake:45 (find_package_handle_standard_args)
  /agent/_work/1/s/scripts/buildsystems/vcpkg.cmake:857 (_find_package)
  cmake/SeastarDependencies.cmake:107 (find_package)
  CMakeLists.txt:309 (seastar_find_dependencies)
```

**Comment by Balnian** [comment]
This will happen when you didn't installed the required package as stated in the console output when doing `vcpkg install seastar`
```bash
-- Seastar currently requires the following libraries from the system package manager:
    libpciaccess
    xfslibs
    libgnutls28
    libsctp
    systemtap-sdt
    libtool
    valgrind 

These can be installed on Ubuntu systems via apt-get install libpciaccess-dev xfslibs-dev libgnutls28-dev libsctp-dev systemtap-sdt-dev libtool valgrind
```

**Comment by JackBoosY** [comment]
When building `seastar:x64-linux`:
```
[4/87] /usr/bin/c++ -DFMT_LOCALE -DSEASTAR_API_LEVEL=2 -DSEASTAR_DEBUG -DSEASTAR_DEBUG_SHARED_PTR -DSEASTAR_DEFAULT_ALLOCATOR -DSEASTAR_HAS_MEMBARRIER -DSEASTAR_HAVE_ASAN_FIBER_SUPPORT -DSEASTAR_HAVE_HWLOC -DSEASTAR_HAVE_LZ4_COMPRESS_DEFAULT -DSEASTAR_HAVE_NUMA -DSEASTAR_SHUFFLE_TASK_QUEUE -DSEASTAR_TYPE_ERASE_MORE -I/home/work/15838/vcpkg/buildtrees/seastar/src/ar-20.05.0-ebf2351e9f.clean/include -Igen/include -isystem /home/work/15838/vcpkg/installed/x64-linux/include -fPIC -g -fvisibility=hidden -UNDEBUG -Wall -Werror -Wno-array-bounds -Wno-error=deprecated-declarations -gz -std=gnu++17 -U_FORTIFY_SOURCE -Wno-maybe-uninitialized -DSEASTAR_SSTRING -Wno-error=unused-result -fstack-clash-protection -fsanitize=address -fsanitize=undefined -fno-sanitize=vptr -MD -MT CMakeFiles/seastar_perf_testing.dir/tests/perf/perf_tests.cc.o -MF CMakeFiles/seastar_perf_testing.dir/tests/perf/perf_tests.cc.o.d -o CMakeFiles/seastar_perf_testing.dir/tests/perf/perf_tests.cc.o -c /home/work/15838/vcpkg/buildtrees/seastar/src/ar-20.05.0-ebf2351e9f.clean/tests/perf/perf_tests.cc
FAILED: CMakeFiles/seastar_perf_testing.dir/tests/perf/perf_tests.cc.o
/usr/bin/c++ -DFMT_LOCALE -DSEASTAR_API_LEVEL=2 -DSEASTAR_DEBUG -DSEASTAR_DEBUG_SHARED_PTR -DSEASTAR_DEFAULT_ALLOCATOR -DSEASTAR_HAS_MEMBARRIER -DSEASTAR_HAVE_ASAN_FIBER_SUPPORT -DSEASTAR_HAVE_HWLOC -DSEASTAR_HAVE_LZ4_COMPRESS_DEFAULT -DSEASTAR_HAVE_NUMA -DSEASTAR_SHUFFLE_TASK_QUEUE -DSEASTAR_TYPE_ERASE_MORE -I/home/work/15838/vcpkg/buildtrees/seastar/src/ar-20.05.0-ebf2351e9f.clean/include -Igen/include -isystem /home/work/15838/vcpkg/installed/x64-linux/include -fPIC -g -fvisibility=hidden -UNDEBUG -Wall -Werror -Wno-array-bounds -Wno-error=deprecated-declarations -gz -std=gnu++17 -U_FORTIFY_SOURCE -Wno-maybe-uninitialized -DSEASTAR_SSTRING -Wno-error=unused-result -fstack-clash-protection -fsanitize=address -fsanitize=undefined -fno-sanitize=vptr -MD -MT CMakeFiles/seastar_perf_testing.dir/tests/perf/perf_tests.cc.o -MF CMakeFiles/seastar_perf_testing.dir/tests/perf/perf_tests.cc.o.d -o CMakeFiles/seastar_perf_testing.dir/tests/perf/perf_tests.cc.o -c /home/work/15838/vcpkg/buildtrees/seastar/src/ar-20.05.0-ebf2351e9f.clean/tests/perf/perf_tests.cc
c++: error: unrecognized command line option ‘-fstack-clash-protection’; did you mean ‘-fstack-protector’?
```

**Comment by Balnian** [comment]
Never had this on my side (both personnal computer and clean docker ubuntu builds), quick googling is pointing me to an "old" version of gcc.
@JackBoosY can you confirm the version of gcc that is used?

**Comment by Balnian** [comment]
I've just confirmed that this flag is checked for support before being enabled. [link ](https://github.com/scylladb/seastar/blob/ea5e529f300dce1297adc615f5e7d847571a04da/CMakeLists.txt#L975-L979) to relevant snippet 
Once I have the compiler version and a local repro, I'll check what's the best fix

**Comment by JackBoosY** [comment]
@Balnian 
```sh
root@usr:/home/usr# gcc -v
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/7/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Ubuntu 7.5.0-3ubuntu1~18.04' --with-bugurl=file:///usr/share/doc/gcc-7/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-7 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-objc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 7.5.0 (Ubuntu 7.5.0-3ubuntu1~18.04)
root@usr:/home/usr#
```

**Comment by BillyONeal** [comment]
The author is still working on this for unrelated-to-vm-update reasons, so dropping requires:vm-update for now. (I'm nervous about a lot of those dependencies: valgrind seems like a dev dependency rather than something users should need, and gnutls seems to compete with vcpkg's default of using openssl as the TLS implementation. But I don't need to answer those questions if there are other blocking things that cna be resolved.)

@Balnian if you're still working on this and have those other issues dealt with please @ me.

**Comment by JackBoosY** [comment]
Pinging @Balnian for response. Is work still being done for this PR?

**Comment by Balnian** [comment]
I don't have time to work on this anymore for the foreseeable future...
btw I was also surprised by the dependencies @BillyONeal but it wouldn't build without them.

I'm not sure what you usually do in these case, do we close it? Do we just let it here for someone to pick up?

**Comment by BillyONeal** [comment]
> I'm not sure what you usually do in these case, do we close it? Do we just let it here for someone to pick up?

I would close it; someone who wants to start from this work can always work from your branch; having the PR outstanding means we're spending 'cycles' reviewing it, even if only in seeing the title in lists.

**Comment by JackBoosY** [comment]
This PR has been inactive for a long time, please reopen it if there is any progress.

**Pull request closed without merging**